### PR TITLE
Add typings and fix exposed bugs

### PR DESCRIPTION
**Pull request by dotted**
Explicit type documentation only added if typescript would complain without it.

Bonus meme: Add debug config for VS Code so logging gets captured during debug

### CODE REVIEW

**Comment by dotted** [comment]
I have only validated that unit tests are green.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (linusred) [inline .vscode/launch.json]: We can git ignore this file, it seems pretty generic and IDE specific. If everyone working on a project added their IdE configuration files, we'd quickly have a large number of IDE configuration files.

File: jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (linusred) [inline jsconfig.json]: This also seems IDE specific, should be git ignored.

Comment (linusred) [reply jsconfig.json]: If we really want to, we can include them in a wiki help page about visual studio code specific configurations

File: lib/commands/base-punishment-helper.js [hunk 3]
```diff
@@ -6,14 +6,14 @@ function basePunishmentHelper(input, defaultDuration, overRideDuration) {
   const matched = /(\d+[hmds])?\s?(\w+)/.exec(input);
   const userToPunish = _.get(matched, 2, '').toLowerCase();
   let parsedDuration = 0;
-  if (parsedDuration !== false) {
-    parsedDuration = _.get(matched, 1, '').toLowerCase();
-    switch (parsedDuration) {
+  if (!overRideDuration) {
```
Comment (linusred) [inline lib/commands/base-punishment-helper.js]: Nice catch, how about !== undefined though. A value of 0 here triggers the wrong logic.

File: lib/commands/implementations/unban.js [hunk 4]
```diff
@@ -9,8 +9,8 @@ function unban(input, services) {
   if (_.isEmpty(userToUnban)) {
     return new CommandOutput(new Error('User was empty.'), `Something did not work. ${userToUnban} not unbanned.`);
   }
-  services.punishmentStream.write(makeUnban(userToUnban, `Unbanning ${userToUnban}`));
-  return new CommandOutput(null, null, null);
+  services.punishmentStream.write(makeUnban(userToUnban));
```
Comment (linusred) [inline lib/commands/implementations/unban.js]: This is left over from some refactoring I did, I think the change here should actually be to just get rid of the output from both makeUnmute and makeUnban, and have no output in unmute or unban. Neither twitch nor DGG supports reasons for unban/unmute, and they also both automatically output to the chat, so ouputting twice looks bad.

File: lib/commands/base-punishment-helper.js (hunk 3 above)
Comment (dotted) [reply lib/commands/base-punishment-helper.js]: Good point

File: .vscode/launch.json (hunk 1 above)
Comment (dotted) [reply .vscode/launch.json]: The important part here is `"outputCapture": "std"` as this needs to be manually added, the rest is auto generated. Could be on a wiki page.

File: jsconfig.json (hunk 2 above)
Comment (dotted) [reply jsconfig.json]: Technically it's `tsserver` specific, so any IDE using that (which would be most, if not all supporting TypeScript) should support this file.

That doesn't really address your point, but I'd support putting it in a wiki.

**Pull request merged**